// @astrojs/vue/dist/index.js
import path from "node:path";
import vue from "@vitejs/plugin-vue";
function getRenderer() {
  return {
    name: "@astrojs/vue",
    clientEntrypoint: "@astrojs/vue/client.js",
    serverEntrypoint: "@astrojs/vue/server.js"
  };
}
function getJsxRenderer() {
  return {
    name: "@astrojs/vue (jsx)",
    clientEntrypoint: "@astrojs/vue/client.js",
    serverEntrypoint: "@astrojs/vue/server.js",
    jsxImportSource: "vue",
    jsxTransformOptions: async () => {
      const jsxPlugin = (await import("@vue/babel-plugin-jsx")).default;
      return {
        plugins: [jsxPlugin]
      };
    }
  };
}
function virtualAppEntrypoint(options) {
  const virtualModuleId = "virtual:@astrojs/vue/app";
  const resolvedVirtualModuleId = "\0" + virtualModuleId;
  let isBuild;
  let root;
  return {
    name: "@astrojs/vue/virtual-app",
    config(_, { command }) {
      isBuild = command === "build";
    },
    configResolved(config) {
      root = config.root;
    },
    resolveId(id) {
      if (id == virtualModuleId) {
        return resolvedVirtualModuleId;
      }
    },
    load(id) {
      if (id === resolvedVirtualModuleId) {
        if (options?.appEntrypoint) {
          const appEntrypoint = options.appEntrypoint.startsWith(".") ? path.resolve(root, options.appEntrypoint) : options.appEntrypoint;
          return `import * as mod from ${JSON.stringify(appEntrypoint)};
						
export const setup = (app) => {
	if ('default' in mod) {
		mod.default(app);
	} else {
		${!isBuild ? `console.warn("[@astrojs/vue] appEntrypoint \`" + ${JSON.stringify(
            appEntrypoint
          )} + "\` does not export a default function. Check out https://docs.astro.build/en/guides/integrations-guide/vue/#appentrypoint.");` : ""}
	}
}`;
        }
        return `export const setup = () => {};`;
      }
    }
  };
}
async function getViteConfiguration(options) {
  const config = {
    optimizeDeps: {
      include: ["@astrojs/vue/client.js", "vue"],
      exclude: ["@astrojs/vue/server.js", "virtual:@astrojs/vue/app"]
    },
    plugins: [vue(options), virtualAppEntrypoint(options)],
    ssr: {
      external: ["@vue/server-renderer"],
      noExternal: ["vuetify", "vueperslides", "primevue"]
    }
  };
  if (options?.jsx) {
    const vueJsx = (await import("@vitejs/plugin-vue-jsx")).default;
    const jsxOptions = typeof options.jsx === "object" ? options.jsx : void 0;
    config.plugins?.push(vueJsx(jsxOptions));
  }
  return config;
}
function src_default(options) {
  return {
    name: "@astrojs/vue",
    hooks: {
      "astro:config:setup": async ({ addRenderer, updateConfig }) => {
        addRenderer(getRenderer());
        if (options?.jsx) {
          addRenderer(getJsxRenderer());
        }
        updateConfig({ vite: await getViteConfiguration(options) });
      }
    }
  };
}
export {
  src_default as default
};
